DNN Platform/Library/Security/Permissions/PermissionController.cs

```diff
@@ -215,6 +270,39 @@ private static IEnumerable<PermissionInfo> GetPermissions()
                 c => CBO.FillCollection<PermissionInfo>(Provider.ExecuteReader("GetPermissions")));
         }
 
+        private static IEnumerable<PermissionInfo> GetPermissionsByFolderEnumerable()
+        {
+            return GetPermissions().Where(p => p.PermissionCode == "SYSTEM_FOLDER");
+        }
+
+        private static IEnumerable<PermissionInfo> GetPermissionsByPortalDesktopModuleEnumerable()
+        {
+            return GetPermissions().Where(p => p.PermissionCode == "SYSTEM_DESKTOPMODULE");
+        }
+
+        private static IEnumerable<PermissionInfo> GetPermissionsByTabEnumerable()
+        {
+            return GetPermissions().Where(p => p.PermissionCode == "SYSTEM_TAB");
+        }
+
+        private static IEnumerable<PermissionInfo> GetPermissionByCodeAndKeyEnumerable(string permissionCode, string permissionKey)
+        {
+            return GetPermissions().Where(p => p.PermissionCode.Equals(permissionCode, StringComparison.InvariantCultureIgnoreCase)
+                                               && p.PermissionKey.Equals(permissionKey, StringComparison.InvariantCultureIgnoreCase));
+        }
+
+        private static IEnumerable<PermissionInfo> GetPermissionsByModuleDefIdEnumerable(int moduleDefId)
+        {
+            return GetPermissions().Where(p => p.ModuleDefID == moduleDefId);
+        }
+
+        private static IEnumerable<PermissionInfo> GetPermissionsByModuleEnumerable(int moduleId, int tabId)
+        {
+            var module = ModuleController.Instance.GetModule(moduleId, tabId, false);
+
+            return GetPermissions().Where(p => p.ModuleDefID == module.ModuleDefID || p.PermissionCode == "SYSTEM_MODULE_DEFINITION");

```

Comment:
Let's not capture the `ModuleInfo` instance in the closure.

```suggestion
            var module = ModuleController.Instance.GetModule(moduleId, tabId, false);
            var moduleDefId = module.ModuleDefID;

            return GetPermissions().Where(p => p.ModuleDefID == moduleDefId || p.PermissionCode == "SYSTEM_MODULE_DEFINITION");

```